### PR TITLE
allow mvi queries if gender is nil (dslogon users)

### PR DESCRIPTION
**Pull request by kreek**
Fixes https://github.com/department-of-veterans-affairs/vets-api/issues/564, in tests some users are found without gender, but MVI team lists it as a required field.

### CODE REVIEW

**Comment by crwallace** [comment]
So this will mean that the user will com back with no data from MVI and we will show them the "You aren't found" message?

**Comment by kreek** [comment]
@crwallace before we didn't query without gender, now we will, they might be found by MVI without gender (I could find some users), but yes if they're not found they'll get the "You aren't found" message.

File: app/models/mvi.rb [hunk 1]
```diff
@@ -68,16 +69,16 @@ def select_source_id(correlation_id)
...
+    @memoized_response ||= response || query_and_cache_response
   end
 
   def mvi_service
     @service ||= MVI::ServiceFactory.get_service(mock_service: ENV['MOCK_MVI_SERVICE'])
   end
 
   def create_message
-    raise Common::Exceptions::ValidationErrors, @user unless @user.valid?
+    raise Common::Exceptions::ValidationErrors, @user unless @user.valid?(:loa3?)
```
Comment (omgitsbillryan) [inline app/models/mvi.rb]: Is this supposed to perform the LOA3 validation? I don't think it is doing so:
```
2.3.0 :001 > u = User.new(uuid: "123456", email: "bill@aol.com", loa: {highest: 1, current: 1})
 => #<User:0x000000023b1fd0 @uuid="123456", @email="bill@aol.com", @loa={:highest=>1, :current=>1}, @first_name=nil, @middle_name=nil, @last_name=nil, @gender=nil, @birth_date=nil, @zip=nil, @ssn=nil, @last_signed_in=nil, @mhv_last_signed_in=nil, @persisted=false> 
2.3.0 :002 > u.valid?
 => true 
2.3.0 :003 > u.valid?(:loa3)
 => true 
2.3.0 :004 >
```

Coincidentally, I'm doing something very similar but I had to change this inside User.rb from:
```ruby
with_options unless: :loa1? do |user|
```
to this,
```ruby
with_options(on: :loa3_user) do |user|
```
and then I could call:
```ruby
@user.valid?             # perform only LOA 1 valiation
@user.valid?(:loa3_user) # perform LOA 1 & LOA 3 validation
```

File: lib/mvi/messages/find_candidate_message.rb [hunk 2]
```diff
@@ -75,7 +75,7 @@ def build_query_by_parameter
 
       def build_parameter_list
         el = element('parameterList')
-        el << build_gender
+        el << build_gender unless @gender.nil?
```
Comment (omgitsbillryan) [inline lib/mvi/messages/find_candidate_message.rb]: Perhaps `.blank?` is safer so that we also account for ''.

**Pull request merged**